CodeGeneration/Sources/generate-swiftsyntax/templates/swiftparser/ParserTokenSpecSetFile.swift

```diff
@@ -0,0 +1,78 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+let parserTokenSpecSetFile = SourceFileSyntax(leadingTrivia: copyrightHeader) {
+  DeclSyntax("@_spi(RawSyntax) import SwiftSyntax")
+
+    for layoutNode in SYNTAX_NODES.compactMap(\.layoutNode) {
+        for child in layoutNode.children {
+          if case let .token(choices, _, _) = child.kind, choices.count > 1 {
+              try! ExtensionDeclSyntax("extension \(raw: layoutNode.kind.rawType)") {
```

Comment:
Is there any reason why you chose to generate the `TokenSpecSet` in the `Raw` types instead of the non-raw types? I’m thinking that if we define them on the public types, might make the `TokenSpecSet` public in there future as they could also be useful for clients to know which kind of tokens a child may contain.